tests/integration/ha_tests/test_ha.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import time
+
+import pytest
+import pytest_asyncio
+from pytest_operator.plugin import OpsTest
+
+from tests.integration.ha_tests.helpers import (
+    count_primaries,
+    deploy_and_scale_application,
+    deploy_and_scale_mongodb,
+    fetch_replica_set_members,
+    get_application_name,
+    get_mongo_client,
+    get_process_pid,
+    get_replica_set_primary,
+    get_units_hostnames,
+    mongod_ready,
+    relate_mongodb_and_application,
+    send_signal_to_pod_container_process,
+)
+from tests.integration.helpers import APP_NAME
+
+MONGODB_CONTAINER_NAME = "mongod"
+MONGOD_PROCESS_NAME = "mongod"
+MEDIAN_REELECTION_TIME = 12
+TEST_DB = "continuous_writes_database"
+TEST_COLLECTION = "test_collection"
+
+
+@pytest_asyncio.fixture
+async def continuous_writes(ops_test: OpsTest) -> None:
+    """Starts continuous writes to the MySQL cluster for a test and clear the writes at the end."""
+    application_name = await get_application_name(ops_test, "application")
+
+    application_unit = ops_test.model.applications[application_name].units[0]
+
+    clear_writes_action = await application_unit.run_action("clear-continuous-writes")
+    await clear_writes_action.wait()
+
+    start_writes_action = await application_unit.run_action("start-continuous-writes")
+    await start_writes_action.wait()
+
+    yield
+
+    clear_writes_action = await application_unit.run_action("clear-continuous-writes")
+    await clear_writes_action.wait()
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    """Build and deploy three units of MongoDB and one test unit."""
+    # it is possible for users to provide their own cluster for HA testing. Hence check if there
+    # is a pre-existing cluster.
+    mongodb_application_name = await get_application_name(ops_test, APP_NAME)
+    if not mongodb_application_name:
+        mongodb_application_name = await deploy_and_scale_mongodb(ops_test)
```

Comment:
just Q: is it possible to run this test against currently running MongoDB k8s charm?

Reply:
Yes, that's how I do it locally to avoid rebuilding the charm on each run.